From 9a8682bacebe60298dbb6677f78c6533a8e15736 Mon Sep 17 00:00:00 2001
From: Srikanth Thokala <srikanth.thokala@xilinx.com>
Date: Fri, 20 Sep 2013 12:29:33 +0530
Subject: [PATCH 139/628] cdma: xilinx: Replace custom my_log() function with
 fls()

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit 37d31af1dfecc778e8a20769b37f97107db3493b

Replace custom my_log() function with fls()

Signed-off-by: Srikanth Thokala <sthokal@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/dma/xilinx/xilinx_axicdma.c |   20 +-------------------
 1 files changed, 1 insertions(+), 19 deletions(-)

diff --git a/drivers/dma/xilinx/xilinx_axicdma.c b/drivers/dma/xilinx/xilinx_axicdma.c
index 9824df7..ee01bb8 100644
--- a/drivers/dma/xilinx/xilinx_axicdma.c
+++ b/drivers/dma/xilinx/xilinx_axicdma.c
@@ -826,24 +826,6 @@ static int xilinx_cdma_device_control(struct dma_chan *dchan,
 	return -ENXIO;
 }
 
-/*
- * Logarithm function to compute alignment shift
- *
- * Only deals with value less than 4096.
- */
-static int my_log(int value)
-{
-	int i = 0;
-	while ((1 << i) < value) {
-		i++;
-
-		if (i >= 12)
-			return 0;
-	}
-
-	return i;
-}
-
 static void xilinx_cdma_free_channels(struct xilinx_cdma_device *xdev)
 {
 
@@ -931,7 +913,7 @@ static int xilinx_cdma_chan_probe(struct xilinx_cdma_device *xdev,
 	chan->common.private = (void *)&(chan->private);
 
 	if (!chan->has_dre)
-		xdev->common.copy_align = my_log(width);
+		xdev->common.copy_align = fls(width - 1);
 
 	chan->dev = xdev->dev;
 	xdev->chan = chan;
-- 
1.7.5.4

